### PR TITLE
Add contact preferences to messages page [#177124054]

### PR DESCRIPTION
**Pull request by bytheway875**
Add a list of contact preferences to the messages page so that greeters and vita users dont need to tab back and forth to see this critical messaging-related information.

### CODE REVIEW

File: app/views/hub/messages/index.html.erb [hunk 1]
```diff
@@ -32,10 +32,31 @@
     <ul class="message-list" data-js="messages-pub-sub" data-client-id="<%=@client.id%>"></ul>
   <% end %>
   <div class="hub-section">
-    <%= form_with model: @outgoing_text_message, url: hub_client_outgoing_text_messages_path(client: @client), local: true, builder: VitaMinFormBuilder, method: "post", html: {class: 'text-message-form'} do |f| %>
-      <%= f.cfa_input_field :body, t(".text_message_form.label") %>
-      <%= f.hidden_field :client_id %>
-      <%= f.submit t(".text_message_form.submit"), class: "button #{"button--disabled" if @client.sms_phone_number.blank?}", disabled: @client.sms_phone_number.blank? %>
+    <div class="communication-preferences">
+      <% if !@client&.intake.email_address.present? && !@client&.intake.sms_phone_number.present? %>
```
Comment (embarnard) [inline app/views/hub/messages/index.html.erb:36]: I think `unless @client&.intake.email_address.present? && @client&.intake.sms_phone_number.present?` would be a little more clear

File: spec/controllers/hub/messages_controller_spec.rb [hunk 2]
```diff
@@ -28,6 +28,60 @@
         end
       end
 
+      context "viewing communication preferences" do
+        render_views
+        context "without way to communicate" do
+          before do
+            intake.update(sms_notification_opt_in: "no", email_notification_opt_in: "no", email_address: "", sms_phone_number: nil)
```
Comment (embarnard) [inline spec/controllers/hub/messages_controller_spec.rb:35]: why do we save blank email addresses as empty strings and blank sms phone numbers as nil?

**Review by embarnard — APPROVED** [review]
Looks good! Left a few comments but nothing blocking :)

File: spec/controllers/hub/messages_controller_spec.rb (hunk 2 above)
Comment (bytheway875) [reply spec/controllers/hub/messages_controller_spec.rb:35]: It's a validation problem -- the create client form is allowing people to send in empty emails but click that they are opted into email. That's funky, but i wanted to replicate that funky behavior in the tests to make sure this catches that case.

File: app/views/hub/messages/index.html.erb (hunk 1 above)
Comment (bytheway875) [reply app/views/hub/messages/index.html.erb:36]: good call! thank you.

Comment (bytheway875) [reply app/views/hub/messages/index.html.erb:36]: BUT, just a clarification that the opposite of `if !x && !y` would need to be `unless x || y`

**Pull request merged**